library/src/interactor_impl.cxx

```diff
@@ -408,6 +392,33 @@ class interactor_impl::internals
     this->VTKInteractor->ExitCallback();
   }
 
+  void AnimateCameraTransition(const std::function<void(camera&, double)>& update_camera)
+  {
+    window& win = this->Window;
+    camera& cam = win.getCamera();
+    const int duration = this->TransitionDuration;
+
+    if (duration > 0)
+    {
+      // TODO can we disable inputs so we don't queue key presses while the animation is running?
```

Comment:
there is a flag to disable mouse movement in the interactor, maybe look at this and see it should be expanded/copied into something that fits what you needd ?

Reply:
Would it be fine to leave it as is for now and deal with it when it actually comes up?

Right now the animation is only triggered from mouse clicks which seems to be fine, the problematic case is with key presses: when the key is held down events get queued faster than the animation plays out and they keep going after the key has been released until all the queued key presses are processes.

I believe that will eventually show up when implementing animation for #14

Reply:
> Would it be fine to leave it as is for now and deal with it when it actually comes up?

Yes, no worries

Reply:
please rework the comment not into a question but into a potential action to take in the future